homeassistant/components/renault/binary_sensor.py

```diff
@@ -59,10 +61,39 @@ class RenaultBinarySensor(
     @property
     def is_on(self) -> bool | None:
         """Return true if the binary sensor is on."""
-        if (data := self._get_data_attr(self.entity_description.on_key)) is None:
-            return None
 
-        return data == self.entity_description.on_value
+        if self.entity_description.value_lambda is not None:
+            return self.entity_description.value_lambda(self)
+        if self.entity_description.on_key is not None:
+            if (data := self._get_data_attr(self.entity_description.on_key)) is None:
+                return None
+
+            return data == self.entity_description.on_value
+        return None
+
+
+def _plugged_in_value_lambda(self: RenaultBinarySensor) -> bool | None:
+    """Return true if the vehicle is plugged in."""
+
+    data = self.coordinator.data
+    plug_status = data.get_plug_status() if data else None
+
+    if plug_status is None:
+        charging_status = data.get_charging_status() if data else None
+        if charging_status is not None and charging_status in [
+            ChargeState.CHARGE_IN_PROGRESS,
+            ChargeState.WAITING_FOR_CURRENT_CHARGE,
+            ChargeState.CHARGE_ENDED,
+            ChargeState.V2G_CHARGING_NORMAL,
+            ChargeState.V2G_CHARGING_WAITING,
+            ChargeState.V2G_DISCHARGING,
+            ChargeState.WAITING_FOR_A_PLANNED_CHARGE,
+        ]:
+            return True
```

Comment:
Maybe it's best to make constant set of ChargeState:
```python
_PLUG_FROM_CHARGE_STATUS: set[ChargeState] = {
    ChargeState.CHARGE_IN_PROGRESS,
    ChargeState.WAITING_FOR_CURRENT_CHARGE,
    ...
}
```
```suggestion
    charging_status = data.get_charging_status() if data else None
    return charging_status in _PLUG_FROM_CHARGE_STATUS if charging_status else None
```